tools/analysis_tools/benchmark.py

```diff
@@ -98,15 +108,34 @@ def main():
 
         if i >= num_warmup:
             pure_inf_time += elapsed
-            if (i + 1) % args.log_interval == 0:
+            if (i + 1) % log_interval == 0:
                 fps = (i + 1 - num_warmup) / pure_inf_time
-                print(f'Done image [{i + 1:<3}/ 2000], fps: {fps:.1f} img / s')
+                print(
+                    f'Done image [{i + 1:<3}/ {max_iter}], '
+                    f'fps: {fps:.1f} img / s',
+                    flush=True)
 
-        if (i + 1) == 2000:
-            pure_inf_time += elapsed
+        if (i + 1) == max_iter:
             fps = (i + 1 - num_warmup) / pure_inf_time
-            print(f'Overall fps: {fps:.1f} img / s')
+            print(f'Overall fps: {fps:.1f} img / s', flush=True)
             break
+    return round(fps, 1)
```

Comment:
1. Considering keep the float number and let users decide whether to round it after they obtain the data.
2. May also return the times per image, which is also common in speed comparison.